### ISSUE TITLE
[apex] New Rule: Annotations should be in pascal case

### ISSUE THREAD

**Issue by mitchspano** [pmd/pmd#5650]
**Proposed Rule Name:** Annotations should be in pascal case.

**Proposed Category:** Code Style

**Description:** This rule enforces that Apex annotations should be written in PascalCase. Apex, while case-insensitive, benefits from a consistent code style to improve readability and maintainability. Enforcing PascalCase for annotations aligns with the established convention for classes and methods, reducing ambiguity and promoting a unified coding standard. Furthermore, Apex currently does not support custom annotations; therefore, the set of allowable annotations is fixed and known at compile time, ensuring that this rule can always be reliably checked.

The current set of permitted annotations is published [here](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation.htm).

**Code Sample:** 

```java
// Incorrect:
@istest
private static void fooShouldBar() {
  //...
}

// Correct:
@IsTest
private static void fooShouldBar() {
  //...
}

//Incorrect:
@testvisible
private boolean doSomething = false;


//Correct:
@TestVisible
private boolean doSomething = false;
```
